ckanext/stadtzhtheme/controller.py

```diff
@@ -0,0 +1,180 @@
+from urllib import urlencode
+import logging
+
+from ckan.plugins import toolkit as tk
+import ckan.model as model
+import ckan.logic as logic
+import ckan.lib.plugins
+from ckan.common import c, config, _, request, OrderedDict
+import ckan.lib.helpers as h
+import ckan.lib.search as search
+import ckan.lib.base as base
+from six import string_types
+
+
+import ckan.controllers.group as group
+
+lookup_group_controller = ckan.lib.plugins.lookup_group_controller
+log = logging.getLogger(__name__)
+get_action = logic.get_action
+NotFound = logic.NotFound
+NotAuthorized = logic.NotAuthorized
+ValidationError = logic.ValidationError
+abort = tk.abort
+render = base.render
+
+
+class OgdzhGroupSearchController(group.GroupController):
+    """
+    backporting [https://github.com/ckan/ckan/pull/4032]
+    form CKAN 2.8: Improved search in group read and bulk page
+    TODO: remove this controller overwrite after upgrade to CKAN 2.8
+
+    This controller is copied over from the groups.py controller
+    of CKAN version 2.8. As soon as the update to 2.8 has been
+    done this patch is not necessary any more.
+
+    This overwrite forces CKAN to use the 'fq' parameter
+    instead of the 'q' parameter for filtering by groups
+    in the solr search query.
+    """
```

Comment:
It is just a comment, but let us keep it brief :) A lot of information is mentioned multiple times, maybe shorten it to something like this:
```
Backporting [https://github.com/ckan/ckan/pull/4032] from CKAN 2.8: Improved search in group read and bulk page.

TODO: remove this controller overwrite after upgrade to CKAN 2.8
```